astropy/io/ascii/ui.py

```diff
@@ -271,6 +275,16 @@ def _guess(table, read_kwargs, format, fast_reader):
             # user supplies delimiter="|" but the guess wants to try delimiter=" ",
             # so skip the guess entirely.
             continue
+
+        if guess_kwargs not in filtered_guess_kwargs:
+            filtered_guess_kwargs.append(guess_kwargs)
+
+    # If there are not at least two formats to guess then return no table and
+    # indicate that guessing did not occur.
+    if len(filtered_guess_kwargs) <= 1:
+        return None, False  # dat, guess
```

Comment:
If there is only one format, why should we return `None`? It is possible that an imcomplete set of options (delimiter etc.) is passed into guess and guess substituted default values for other options. In that case,  `_guess` did something and I would expect it to return something.